doc/source/advanced.rst

```diff
@@ -649,7 +649,7 @@ setting the index of a ``DataFrame/Series`` with a ``category`` dtype would conv
 
    df = pd.DataFrame({'A': np.arange(6),
                       'B': list('aabbca')})
-   df['B'] = df['B'].astype('category', categories=list('cab'))
+   df['B'] = df['B'].astype(pd.api.types.CategoricalDtype(list('cab')))
```

Comment:
I would leave this here

Reply:
I'm not a huge fan of the overloading of `.astype` signature, so I removed all instances where we pass through any keyword args. We should still have an example of just `.astype('category')`, with no kwargs. If you want I can change this back though.

Reply:
ok that's fine, it just reads better if you import ``CategoricalDtype``, e.g.

``df['B'].astype(CategoricalDtype(list('cab')))``

Reply:
I suppose we could deprecate passing the ``categories`` thru ``.astype`` as well